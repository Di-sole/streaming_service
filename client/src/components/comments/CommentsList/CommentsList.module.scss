.container {
    display: flex;
    flex-direction: column;

    &__childcomments {
        width: 90%;
        align-self: flex-end;
    }
}