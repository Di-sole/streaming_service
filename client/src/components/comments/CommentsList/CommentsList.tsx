import IComment from '@/models/IComment';
import styles from './CommentsList.module.scss';
import CommentItem from '../CommentItem/CommentItem';

interface CommentsListProps {
    comments: IComment[];
}

const CommentsList = ({ comments }: CommentsListProps) => {
    return (
        <div className={styles.container} data-testid={'commentsList'}>
            {comments.map((comment) => {
                return (
                    <>
                        <CommentItem key={comment.id} comment={comment} />
                        <div className={styles.container__childcomments}>
                            {comment.childComment?.map((child) => (
                                <CommentItem key={child.id} comment={child} />
                            ))}
                        </div>
                    </>
                );
            })}
        </div>
    );
};

export default CommentsList;
