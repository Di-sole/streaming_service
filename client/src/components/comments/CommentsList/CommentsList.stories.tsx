import { Meta, StoryObj } from '@storybook/react';
import './CommentsList.module.scss';
import CommentsList from './CommentsList';
import { commentsTest } from '../../../testAsserts/testItems';

const meta: Meta<typeof CommentsList> = {
    title: 'Information/Comments',
    component: CommentsList,
    decorators: [
        (Story) => (
            <div style={{ width: '800px' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof CommentsList>;

export const Comments: Story = {
    args: {
        comments: commentsTest,
    },
};
