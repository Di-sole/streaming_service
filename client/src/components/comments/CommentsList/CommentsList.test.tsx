import { render, screen } from '@testing-library/react';
import CommentsList from './CommentsList';
import { commentsTest } from '../../../testAsserts/testItems';

import * as reduxHooks from '../../../store/hooks/redux';

jest.mock('../../../store/hooks/redux');
const useDispatch = jest.spyOn(reduxHooks, 'useAppDispatch');
const mockDispatch = jest.fn();

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
            pathname: '',
        };
    },
}));

describe('COMMENTSLIST SNAPSHOTS TESTS', () => {
    test('CommentsList should not be changed', () => {
        useDispatch.mockReturnValue(mockDispatch);
        render(<CommentsList comments={commentsTest} />);

        const id = screen.getByTestId('commentsList');
        expect(id).toMatchSnapshot();
    });
});
