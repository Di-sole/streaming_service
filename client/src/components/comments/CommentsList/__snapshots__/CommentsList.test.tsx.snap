// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMMENTSLIST SNAPSHOTS TESTS CommentsList should not be changed 1`] = `
<div
  class="container"
  data-testid="commentsList"
>
  <div
    class="container"
    data-testid="commentItem"
  >
    <div
      class="container__header"
    >
      <div
        class="container__info"
      >
        <p
          class="container__username"
        >
          Slater
        </p>
        <time
          class="container__date"
          datetime="Sat Jun 10 2023 15:33:19 GMT+0700 (Красноярск, стандартное время)"
        >
          10 июня 2023
        </time>
      </div>
      <div
        class="container"
        data-testid="voteWidget"
      >
        <button
          class="button button_bright button button_dislike"
          type="button"
        >
          <img
            alt="dislike"
            class="button__image"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
        <div
          class="container__votes"
        >
          10
        </div>
        <button
          class="button button_bright button button_like"
          type="button"
        >
          <img
            alt="like"
            class="button__image"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
      </div>
    </div>
    <div
      class="container__content"
    >
      <div
        class="container__text"
      >
        Великий Эверест пытаются покорить многие. Зачем? Как по мне в этом нет смысла. Но все люди разные, именно о тех, кто готов рискнуть и достигнуть поставленной цели фильм и повествует. И он на реальных событиях. Фильм хороший, но не стоит вдохновляться. Не все пройдет гладко при подъеме. В фильме хорошо показаны глупости людей, амбиции вразрез разуму, и беспомощность перед силой природы.
      </div>
    </div>
    <div
      class="container__actions"
    >
      <button
        class="button button_faded container__button"
        type="button"
      >
        comments.reply
      </button>
    </div>
  </div>
  <div
    class="container__childcomments"
  >
    <div
      class="container"
      data-testid="commentItem"
    >
      <div
        class="container__header"
      >
        <div
          class="container__info"
        >
          <p
            class="container__username"
          >
            Игорь
          </p>
          <time
            class="container__date"
            datetime="Sat Jun 11 2023 15:33:19 GMT+0700 (Красноярск, стандартное время)"
          >
            11 июня 2023
          </time>
        </div>
        <div
          class="container"
          data-testid="voteWidget"
        >
          <button
            class="button button_bright button button_dislike"
            type="button"
          >
            <img
              alt="dislike"
              class="button__image"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
          <div
            class="container__votes"
          >
            10
          </div>
          <button
            class="button button_bright button button_like"
            type="button"
          >
            <img
              alt="like"
              class="button__image"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
        </div>
      </div>
      <div
        class="container__content"
      >
        <div
          class="container__text"
        >
          Ты прав!
        </div>
      </div>
    </div>
    <div
      class="container"
      data-testid="commentItem"
    >
      <div
        class="container__header"
      >
        <div
          class="container__info"
        >
          <p
            class="container__username"
          >
            Гость
          </p>
          <time
            class="container__date"
            datetime="Sat Jun 11 2023 15:33:19 GMT+0700 (Красноярск, стандартное время)"
          >
            11 июня 2023
          </time>
        </div>
        <div
          class="container"
          data-testid="voteWidget"
        >
          <button
            class="button button_bright button button_dislike"
            type="button"
          >
            <img
              alt="dislike"
              class="button__image"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
          <div
            class="container__votes"
          >
            10
          </div>
          <button
            class="button button_bright button button_like"
            type="button"
          >
            <img
              alt="like"
              class="button__image"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
        </div>
      </div>
      <div
        class="container__content"
      >
        <div
          class="container__text"
        >
          Точно
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
    data-testid="commentItem"
  >
    <div
      class="container__header"
    >
      <div
        class="container__info"
      >
        <p
          class="container__username"
        >
          lanaalmaty
        </p>
        <time
          class="container__date"
          datetime="Sat Jun 10 2023 15:33:19 GMT+0700 (Красноярск, стандартное время)"
        >
          10 июня 2023
        </time>
      </div>
      <div
        class="container"
        data-testid="voteWidget"
      >
        <button
          class="button button_bright button button_dislike"
          type="button"
        >
          <img
            alt="dislike"
            class="button__image"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
        <div
          class="container__votes"
        >
          10
        </div>
        <button
          class="button button_bright button button_like"
          type="button"
        >
          <img
            alt="like"
            class="button__image"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
      </div>
    </div>
    <div
      class="container__content"
    >
      <div
        class="container__text"
      >
        Фильм классный! Но здесь все забыли главного героя. Анатолия Букреева. Ведь он свою группу все таки вывел из бури и все живы!
      </div>
    </div>
    <div
      class="container__actions"
    >
      <button
        class="button button_faded container__button"
        type="button"
      >
        comments.reply
      </button>
    </div>
  </div>
  <div
    class="container__childcomments"
  />
</div>
`;
